include/cuco/static_multimap.cuh

```diff
@@ -959,6 +959,37 @@ class static_multimap {
       OutputIt output_begin,
       KeyEqual key_equal = KeyEqual{}) noexcept;
 
+    /**
+     * @brief Retrieves all the matches of a given pair contained in multimap without using shared
+     * memory buffer
+     *
+     * For pair `p`, if pair_equal(p, slot[j]) returns true, copies `p` to unspecified locations
+     * in `[probe_output_begin, probe_output_end)` and copies slot[j] to unspecified locations in
+     * `[contained_output_begin, contained_output_end)`.
+     *
+     * @tparam atomicT Type of atomic storage
+     * @tparam OutputIt1 Device accessible output iterator whose `value_type` is constructible from
+     * `InputIt`s `value_type`.
+     * @tparam OutputIt2 Device accessible output iterator whose `value_type` is constructible from
+     * the map's `value_type`.
+     * @tparam PairEqual Binary callable type
+     * @param probing_cg The Cooperative Group used to retrieve
+     * @param pair The pair to search for
+     * @param num_matches Size of the output sequence
+     * @param probe_output_begin Beginning of the output sequence of the matched probe pairs
+     * @param contained_output_begin Beginning of the output sequence of the matched contained
+     * pairs
+     * @param pair_equal The binary callable used to compare two pairs for equality
+     */
+    template <typename atomicT, typename OutputIt1, typename OutputIt2, typename PairEqual>
+    __device__ __forceinline__ void pair_retrieve(
+      cooperative_groups::thread_block_tile<ProbeSequence::cg_size> const& probing_cg,
+      value_type const& pair,
+      atomicT* num_matches,
```

Comment:
For this variant, I don't think we want this atomic counter. We just want to write all `n` matches to `[begin, begin + n)`. It's the users responsibility to ensure no other threads will attempt to write to overlapping locations.

Reply:
Do you mean we need a raw counter instead or no counter at all? `num_matches` determines the output size or the output index whenever we find a match. I'm not sure how it works without passing it to the device function.


> We just want to write all n matches to [begin, begin + n).

Are you referring to the host bulk API or device view API?

Reply:
I'm talking about the device API. It shouldn't need to take a counter as a function parameter. It just takes an output iterator and will write all the matches starting at that output iterator.